com.unity.netcode.gameobjects/Runtime/NetworkVariable/Collections/NetworkList.cs

```diff
@@ -73,10 +73,20 @@ public override bool IsDirty()
         public override void WriteDelta(Stream stream)
         {
             using var writer = PooledNetworkWriter.Get(stream);
+
+            if (base.IsDirty())
+            {
```

Comment:
Problably stupid question because I'm misunderstanding this. But this causes the full list to be serialized to all connected clients whenever `IsDirty` is set to true? Wouldn't that happen too often? E.g. when:
- An item is added / removed from the list
- The object becomes visible on any client.

Reply:
On the first point, my understanding is that `NetworkList.Add()` will do:

```
            m_List.Add(item); // add to the underlying container

            var listEvent = new NetworkListEvent<T>()  {
                Type = NetworkListEvent<T>.EventType.Add,
...            }; // create an event

            HandleAddListEvent(listEvent); // handle the event
        }
```
This then does:
```
        private void HandleAddListEvent(NetworkListEvent<T> listEvent)
        {
            m_DirtyEvents.Add(listEvent);
            OnListChanged?.Invoke(listEvent);
        }
```
and as such **does not** mark the list dirty. So, `dirty` means: needs complete refresh, and `m_DirtyEvents` means need incremental changes.

On the second point, yeah. There's a price to pay here. But until AoI is integrated and we start automatically hiding/showing object, I think these will be very infrequent.

Reply:
Okay yeah that makes sense

Reply:
I follow, but would add:
- please add a comment to the effect of "IsDirty means totally dirty, else m_DirtyEvents means incremental
- but I have a hunch this will bite us and should be re-visited